### ISSUE TITLE
Incorrect mapping from grpc code to http code for `canceled request`.

### ISSUE THREAD

**Issue by brijeshvishwanath** [grpc-ecosystem/grpc-gateway#2707]
## 🐛 Bug Report

Incorrect mapping from grpc code to http code for `canceled request`.

The reference for error mapping is followed from:
https://github.com/googleapis/googleapis/blob/942691f8dcf3e521be35d909de9bba3239feb471/google/rpc/code.proto#L40
And the reference claims that http code mapping for CANCELLED grpc code is 499 (Client Closed Request), and not  408 (StatusRequestTimeout).

## To Reproduce

Implement the HTTPStatusFromCode method from the runtime pkg, and see the incorrect mapping for Canceled GRPC code.

## Expected behavior

GRPC Canceled code must be mapped to http code 499 (Client Closed Request), instead of 408 (StatusRequestTimeout).

## Actual Behavior

GRPC Canceled code maps to http 408 (StatusRequestTimeout).

**Comment by johanbrandhorst** [comment]
Huh, I could've sworn I went and checked these. Thank you for your issue and the PR!

**Comment by Rajarshi1998** [comment]
I would like to contribute to fix this issue. Thank you!!

**Comment by johanbrandhorst** [comment]
Hi, please feel free to open a new PR, we haven't heard anything from @brijeshvishwanath for a while.

**Comment by dbeylkhanov** [comment]
humm, .NET doesn't support 499 status code. Thread with discussion here 🧵 https://github.com/dotnet/aspnetcore/issues/41475

**Comment by johanbrandhorst** [comment]
Non-standard or not, we should be consistent with the `code.proto` definitions in our error conversions. I still think we want this change.

**Comment by tech-sumit** [comment]
Hi @johanbrandhorst,
Created PR to contribute this fix, please check.
